packages/core/src/state-management/repository.ts

```diff
@@ -75,7 +75,12 @@ export class Repository {
   ) {
     this.loadingQ = new ExecutionQueue(concurrencyLimit, logger)
     this.executionQ = new ExecutionQueue(concurrencyLimit, logger)
-    this.inmemory = new StateCache(cacheLimit, (state$) => state$.complete())
+    this.inmemory = new StateCache(cacheLimit, (state$) => {
+      if (state$.subscriptionSet.size > 0) {
+        logger.warn(`Stream ${state$.id} evicted from cache while having subscriptions`)
```

Comment:
Once the CAS change is made, this could happen and not actually be a problem.  It's also a log message that most node operators are unlikely to understand what it means or what to do about it. So I think it probably doesn't make sense to show up all the time as a warning, but rather be at the debug level that has to be opted-into for diagnosing issues with the system.  We run all of our hosted nodes with debug logging enabled anyway so we'll see it for our nodes in either case, but I think it makes sense not to fill regular user's logs with warnings over this

Reply:
Thanks for the suggestion!